Python unstable 2021-11-25

jonringer opened the pull request:
###### Motivation for this change
Python update of minor versions.

There seemed to be excessive x86_64-linux hydra capacity. So may as well.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

fabaff committed:
python3Packages.datasette: 0.58.1 -> 0.59

fabaff committed:
wapiti: 3.0.5 -> 3.0.7

fabaff committed:
python3Packages.envoy-reader: allow later httpx releases

fabaff committed:
python3Packages.sanic-auth: support later httpx releases

fabaff committed:
python3Packages.python-socks: 1.2.4 -> 2.0.0

Commit:
python3Packages.jellyfin-apiclient-python: 1.7.2 -> 1.8.0

Commit:
python3Packages.Jinja2: 3.0.2 -> 3.0.3

Commit:
python3Packages.jenkins-job-builder: 3.10.0 -> 3.11.0

Commit:
python3Packages.josepy: 1.10.0 -> 1.11.0

Commit:
python3Packages.jupyter_core: 4.8.1 -> 4.9.1

Commit:
python3Packages.jsonwatch: 0.2.0 -> 0.6.0

Commit:
python3Packages.jupyter_client: 7.0.6 -> 7.1.0

Commit:
python3Packages.jsonpointer: 2.1 -> 2.2

Commit:
python3Packages.jupyterlab: 3.2.3 -> 3.2.4

Commit:
python3Packages.kombu: 5.2.1 -> 5.2.2

Commit:
python3Packages.ledgerblue: 0.1.38 -> 0.1.41

Commit:
python3Packages.labelbox: 3.6.1 -> 3.10.0

Commit:
python3Packages.libcst: 0.3.21 -> 0.3.23

Commit:
python3Packages.k5test: 0.10.0 -> 0.10.1

Commit:
python3Packages.langcodes: 3.2.1 -> 3.3.0

Commit:
python3Packages.Mako: 1.1.5 -> 1.1.6

Commit:
python3Packages.limnoria: 2021.10.9 -> 2021.11.20

Commit:
python3Packages.python-lz4: 3.1.3 -> 3.1.10

Commit:
python3Packages.lxml: 4.6.3 -> 4.6.4-5

Commit:
python3Packages.metakernel: 0.27.5 -> 0.28.1

Commit:
python3Packages.Markups: 3.1.2 -> 3.1.3

Commit:
python3Packages.mlflow: 1.20.2 -> 1.21.0

Commit:
python3Packages.mautrix: 0.11.3 -> 0.12.4

Commit:
python3Packages.mayavi: 4.7.3 -> 4.7.4

Commit:
python3Packages.matplotlib: 3.4.3 -> 3.5.0

Commit:
python3Packages.mysqlclient: 2.0.3 -> 2.1.0

Commit:
python3Packages.msgpack: 1.0.2 -> 1.0.3

Commit:
python3Packages.mocket: 3.10.0 -> 3.10.1

Commit:
python3Packages.natsort: 7.1.1 -> 7.2.0

Commit:
python3Packages.more-itertools: 8.10.0 -> 8.12.0

Commit:
python3Packages.numpy: 1.21.2 -> 1.21.4

Commit:
python3Packages.nipype: 1.6.1 -> 1.7.0

Commit:
python3Packages.nbconvert: 6.2.0 -> 6.3.0

Commit:
python3Packages.notebook: 6.4.4 -> 6.4.6

Commit:
python3Packages.openstacksdk: 0.59.0 -> 0.60.0

Commit:
python3Packages.pbr: 5.7.0 -> 5.8.0

Commit:
python3Packages.paramiko: 2.7.2 -> 2.8.0

Commit:
python3Packages.packet-python: 1.44.1 -> 1.44.2

Commit:
python3Packages.pandas: 1.3.3 -> 1.3.4

Commit:
python3Packages.panel: 0.12.1 -> 0.12.5

Commit:
python3Packages.parsy: 1.3.0 -> 1.4.0

Commit:
python3Packages.pdftotext: 2.2.1 -> 2.2.2

Commit:
python3Packages.Pillow: 8.3.2 -> 8.4.0

Commit:
python3Packages.pex: 2.1.55 -> 2.1.56

Commit:
python3Packages.pip-tools: 6.3.1 -> 6.4.0

Commit:
python3Packages.pc-ble-driver-py: 0.16.1 -> 0.16.2

Commit:
python3Packages.prometheus-client: 0.11.0 -> 0.12.0

Commit:
python3Packages.proto-plus: 1.19.4 -> 1.19.8

Commit:
python3Packages.plumbum: 1.7.0 -> 1.7.1

Commit:
python3Packages.prettytable: 2.2.1 -> 2.4.0

Commit:
python3Packages.plotly: 5.3.1 -> 5.4.0

Commit:
python3Packages.pybids: 0.13.2 -> 0.14.0

Commit:
python3Packages.pudb: 2021.1 -> 2021.2.2

Commit:
python3Packages.psycopg2: 2.9.1 -> 2.9.2

Commit:
python3Packages.pyaml: 21.8.3 -> 21.10.1

Commit:
python3Packages.py4j: 0.10.9.2 -> 0.10.9.3

Commit:
python3Packages.pycparser: 2.20 -> 2.21

Commit:
python3Packages.pydmd: 0.3.3 -> 0.4

Commit:
python3Packages.py: 1.10.0 -> 1.11.0

Commit:
python3Packages.pydeck: 0.7.0 -> 0.7.1

Commit:
python3Packages.pymongo: 3.12.0 -> 3.12.1

Commit:
python3Packages.pyfakefs: 4.5.1 -> 4.5.3

Commit:
python3Packages.pymavlink: 2.4.17 -> 2.4.18

Commit:
python3Packages.pylint: 2.11.1 -> 2.12.1

Commit:
python3Packages.pyfaidx: 0.6.2 -> 0.6.3.1

Commit:
python3Packages.pygit2: 1.7.0 -> 1.7.1

Commit:
python3Packages.pytest-httpx: 0.13.0 -> 0.15.0

Commit:
python3Packages.pymunk: 6.2.0 -> 6.2.1

Commit:
python3Packages.pytest-asyncio: 0.15.1 -> 0.16.0

Commit:
python3Packages.pysam: 0.17.0 -> 0.18.0

Commit:
python3Packages.pytest-doctestplus: 0.11.0 -> 0.11.1

Commit:
python3Packages.python-cinderclient: 8.1.0 -> 8.2.0

Commit:
python3Packages.python-dotenv: 0.19.0 -> 0.19.2

Commit:
python3Packages.pytest-localserver: 0.5.0 -> 0.5.1

Commit:
python3Packages.python-gnupg: 0.4.7 -> 0.4.8

Commit:
python3Packages.python-lsp-server: 1.2.4 -> 1.3.1

Commit:
python3Packages.pyvips: 2.1.15 -> 2.1.16

Commit:
python3Packages.PyWavelets: 1.1.1 -> 1.2.0

Commit:
python3Packages.python-swiftclient: 3.12.0 -> 3.13.0

Commit:
python3Packages.pytwitchapi: 2.5.0 -> 2.5.1

Commit:
python3Packages.pywick: 0.5.6 -> 0.6.5

Commit:
python3Packages.python-manilaclient: 3.0.0 -> 3.1.0

Commit:
python3Packages.reikna: 0.7.5 -> 0.7.6

Commit:
python3Packages.rcssmin: 1.0.6 -> 1.1.0

Commit:
python3Packages.reportlab: 3.6.1 -> 3.6.2

Commit:
python3Packages.r2pipe: 1.6.4 -> 1.6.5

Commit:
python3Packages.qtconsole: 5.1.1 -> 5.2.1

Commit:
python3Packages.rich: 10.12.0 -> 10.14.0

Commit:
python3Packages.rjsmin: 1.1.0 -> 1.2.0

Commit:
python3Packages.robotframework-sshlibrary: 3.7.0 -> 3.8.0

Commit:
python3Packages.rsa: 4.7.2 -> 4.8

Commit:
python3Packages.requests-kerberos: 0.12.0 -> 0.13.0

Commit:
python3Packages.rfc6555: 0.0.0 -> 0.1.0

Commit:
python3Packages.schema-salad: 8.2.20211104054942 -> 8.2.20211116214159

Commit:
python3Packages.s3fs: 2021.10.1 -> 2021.11.0

Commit:
python3Packages.sanic: 21.9.1 -> 21.9.3

Commit:
python3Packages.sagemaker: 2.63.1 -> 2.69.0

Commit:
python3Packages.scikit-hep-testdata: 0.4.10 -> 0.4.11

Commit:
python3Packages.scipy: 1.7.1 -> 1.7.3

jonringer committed:
python3Packages.shap: 0.39.0 -> 0.40.0

Commit:
python3Packages.secp256k1: 0.13.2 -> 0.14.0

jonringer committed:
python3Packages.sentry-sdk: 1.4.3 -> 1.5.0

jonringer committed:
python3Packages.smpplib: 2.2.0 -> 2.2.1

jonringer committed:
python3Packages.snowflake-connector-python: 2.6.2 -> 2.7.1

jonringer committed:
python3Packages.shortuuid: 1.0.1 -> 1.0.8

jonringer committed:
python3Packages.snowballstemmer: 2.1.0 -> 2.2.0

jonringer committed:
python3Packages.simplejson: 3.17.5 -> 3.17.6

jonringer committed:
python3Packages.simple_di: 0.1.2 -> 0.1.4

jonringer committed:
python3Packages.spacy-alignments: 0.8.3 -> 0.8.4

jonringer committed:
python3Packages.soupsieve: 2.2.1 -> 2.3.1

jonringer committed:
python3Packages.sphinxext-opengraph: 0.4.2 -> 0.5.0

jonringer committed:
python3Packages.spyder: 5.1.5 -> 5.2.0

jonringer committed:
python3Packages.spacy: 3.1.3 -> 3.2.0

jonringer committed:
python3Packages.sqlite-utils: 3.17.1 -> 3.19

jonringer committed:
python3Packages.stumpy: 1.9.2 -> 1.10.0

jonringer committed:
python3Packages.strictyaml: 1.5.0 -> 1.6.0

jonringer committed:
python3Packages.stripe: 2.62.0 -> 2.63.0

jonringer committed:
python3Packages.structlog: 21.2.0 -> 21.4.0

jonringer committed:
python3Packages.tifffile: 2021.8.30 -> 2021.11.2

jonringer committed:
python3Packages.transformers: 4.12.2 -> 4.12.5

jonringer committed:
python3Packages.taskw: 1.3.0 -> 1.3.1

jonringer committed:
python3Packages.toolz: 0.11.1 -> 0.11.2

jonringer committed:
python3Packages.trytond: 6.2.0 -> 6.2.1

jonringer committed:
python3Packages.typeguard: 2.12.1 -> 2.13.2

jonringer committed:
python3Packages.ujson: 4.2.0 -> 4.3.0

jonringer committed:
python3Packages.vispy: 0.9.3 -> 0.9.4

jonringer committed:
python3Packages.virtualenv: 20.8.1 -> 20.10.0

jonringer committed:
python3Packages.twentemilieu: 0.3.0 -> 0.4.2

jonringer committed:
python3Packages.types-protobuf: 3.18.0 -> 3.18.1

jonringer committed:
python3Packages.widgetsnbextension: 3.5.1 -> 3.5.2

jonringer committed:
python3Packages.xattr: 0.9.7 -> 0.9.8

jonringer committed:
python3Packages.wled: 0.8.0 -> 0.10.0

jonringer committed:
python3Packages.XStatic-jquery-ui: 1.12.1.1 -> 1.13.0.1

jonringer committed:
python3Packages.wrapt: 1.13.1 -> 1.13.3

jonringer committed:
python3Packages.zope.testrunner: 5.3.0 -> 5.4.0

jonringer committed:
python3Packages.zope.schema: 6.1.0 -> 6.2.0

jonringer committed:
python3Packages.youtube-search-python: 1.5.1 -> 1.5.2

jonringer committed:
python3Packages.zarr: 2.10.2 -> 2.10.3

jonringer committed:
python3Packages.zope.proxy: 4.4.0 -> 4.5.0

thiagokokada commented on pkgs/development/python-modules/boto3/default.nix:
```diff
@@ -13,11 +13,11 @@
 buildPythonPackage rec {
   pname = "boto3";
-  version = "1.18.54"; # N.B: if you change this, change botocore and awscli to a matching version
+  version = "1.20.13"; # N.B: if you change this, change botocore and awscli to a matching version
```
Should we also update awscli :thinking: ?

jonringer replied:
probably, I'll test that next

jonringer replied:
fixed

thiagokokada commented on pkgs/development/python-modules/frozendict/default.nix:
```diff
@@ -8,14 +8,14 @@
 buildPythonPackage rec {
   pname = "frozendict";
-  version = "2.0.7";  # 2.0.6 breaks canonicaljson
+  version = "2.1.0";  # 2.0.6 breaks canonicaljson
```
Can we remove this comment :thinking: ?

jonringer replied:
yea

jonringer replied:
done

jonringer committed:
python3Packages.paramiko: disable tests

Deprecation warnings kill the test suite
https://github.com/pyinvoke/invoke/issues/829

jonringer committed:
Revert "python3Packages.docutils: 0.17.1 -> 0.18.1"

This reverts commit e94af69c7c01067f64bef5bdc0e21cf18677b988.

dotlambda committed:
python3Packages.pylama: 7.7.1 -> 8.0.6

https://github.com/klen/pylama/blob/8.0.6/Changelog

dotlambda committed:
python3Packages.pydocstyle: 5.0.2 -> 6.1.1

dotlambda committed:
python3Packages.trio: don't depend on pylint

dotlambda committed:
python2Packages.pydocstyle: drop

It's not used by any leaf package.

mweinelt committed:
python3Packages.astroid: 2.8.2 -> 2.9.0

mweinelt committed:
python3Packages.pylint: fix tests

mweinelt committed:
python3Packages.aiohttp-wsgi: disable failing test

mweinelt committed:
python3Packages.aiopvapi: 1.6.14 -> unstable-2021-09-07; apply patch to drop loop kwarg

mweinelt committed:
python3Packages.plugwise: 0.15.2 -> 0.15.3

fabaff committed:
python3Packages.websockets: 10.0 -> 10.1

dotlambda commented:
Should we include https://github.com/NixOS/nixpkgs/pull/147524?

dotlambda commented:
There's also https://github.com/NixOS/nixpkgs/pull/145807 but I don't think we want that just yet?

fabaff committed:
python3Packages.screenlogicpy: add pytest-asyncio

fabaff committed:
python3Packages.charset-normalizer: 2.0.5 -> 2.0.8

fabaff committed:
python3Packages.watchdog: disable flaky tests

jonringer committed:
python3Packages.frozendict: disable formatting tests

Too sensitive to interpreter's __repr__ format

jonringer committed:
awscli: 1.20.54 -> 1.22.14

jonringer committed:
python3Packages.botocore: 1.23.13 -> 1.23.14

jonringer committed:
python3Packages.sphinx: 4.2.0 -> 4.3.1

jonringer committed:
python3Packages.fonttools: avoid matplotlib in closure, cyclic dependencies

jonringer committed:
python3Packages.fs: disable tests due to cyclic dependencies with parameterized

jonringer committed:
python3Packages.matplotlib: fix build, disable network fetch of deps

jonringer committed:
python3Packages.toolz: fix tests

jonringer committed:
python3Packages.requests-kerberos: fix build and tests

jonringer committed:
python3Packages.janus: fix missing dep

jonringer committed:
python3Packages.shortuuid: fix tests

jonringer committed:
python3Packages.structlog: fetch from github, fix tests

jonringer committed:
python3Packages.spacy-alignments: fix build, remove old patch

fabaff committed:
python3Packages.homematicip: add support for async_timeout>4

fabaff committed:
python3Packages.google-api-core: disable failing grpc tests

fabaff committed:
python3Packages.google-api-python-client: remove six

fabaff committed:
python3Packages.grpcio-status: init at 1.42.0

fabaff committed:
python3Packages.google-cloud-pubsub: add grpcio-status

fabaff committed:
python3Packages.wled: add missing dependencies

fabaff committed:
python3Packages.pyfaidx: fix build

fabaff committed:
python3Packages.pudb: add missing dependencies

fabaff committed:
python3Packages.pex: switch to flit

fabaff committed:
python3Packages.async_stagger: disable failing test

fabaff committed:
python3Packages.genanki: add missing dependency

fabaff committed:
python3Packages.ephem: 4.1 -> 4.1.1

fabaff committed:
python3Packages.aiofiles: 0.7.0 -> 0.8.0

jonringer committed:
python3Packages.httpx: 0.19.0 -> 0.21.1

jonringer committed:
python3Packages.httpcore: 0.13.7 -> 0.14.3

jonringer committed:
python3Packages.send2trash: 1.5.0 -> 1.8.1b0

jonringer committed:
python3Packages.matplotlib: remove debug lines

jonringer committed:
python3Packages.django_compressor: fix build

jonringer committed:
python3Packages.xdist: disable flakey test

fabaff committed:
python3Packages.respx: 0.17.1 -> 0.19.0

jonringer committed:
python3Packages.androguard: fix build

jonringer commented:
@fabaff can you avoid force pushing to the branch when not necessary.

You deleted a few of my commits

jonringer committed:
python3Packages.frozendict: remove no longer relevant comment

fabaff commented:
@jonringer, sorry, my bad.

fabaff committed:
python3Packages.karton-dashboard: fix build

jonringer committed:
python3Packages.twentemilieu: fix build

fabaff committed:
python3Packages.gdown: add missing dependency

jonringer committed:
python3Packages.libcst: fix build

jonringer committed:
python2Packages.jinja2: disable tests, pytest no longer filters warnings

fabaff commented:
I went for `twentemilieu` and `libcst` too. Looks like this was the start of my questionable actions.

jonringer committed:
python3Packages.imageio: fix tests

jonringer committed:
python3Packages.cytoolz: fix tests

fabaff committed:
python3Packages.elmax: 0.1.3 -> 0.1.5

fabaff committed:
python3Packages.language-data: init at 1.0.1

fabaff committed:
python3Packages.langcodes: add language-data

fabaff committed:
python3Packages.jaraco-context: init at 4.1.1

fabaff committed:
python3Packages.jaraco_text: add jaraco-context

fabaff committed:
flatcam: add packaging

fabaff committed:
corrscope: add packaging

fabaff committed:
python3Packages.rfc6555: switch to GitHub and pytestCheckHook

jonringer committed:
python3Packages.pastescript: disable tests suite, doesn't respect PYTHONPATH

fabaff committed:
python3Packages.adjusttext: add packaging

fabaff committed:
python3Packages.avro: switch to pytestCheckHook

fabaff commented:
I will fix the whitespace issue in ce26888 when I'm sure that nobody else is working.

zowoq commented:
https://github.com/NixOS/nixpkgs/runs/4360946550?check_suite_focus=true

Sorry about the spurious output in addition to the correct whitespace error in the editorconfig check , it's since been reverted.

jonringer committed:
python3Packages.jschema-to-python: init at 1.2.3

jonringer committed:
python3Packages.sarif-om: init at 1.0.4

jonringer committed:
python3Packages.cfn-lint: add missing deps

jonringer commented:
@fabaff I fixed the extra whitespace

fabaff committed:
python3Packages.google-api-core: add grpcio-status

fabaff committed:
python3Packages.google-cloud-access-context-manager: remove stale substituteInPlace

fabaff committed:
python3Packages.apsw: switch to pytestCheckHook, disable failing tests

fabaff committed:
python3Packages.webssh: switch to pytestCheckHook

fabaff committed:
python3Packages.hyperopt: add py4j

jonringer committed:
azure-cli: fix namespacing of dependency

jonringer committed:
python3Packages.azure-keyvault: fix build

jonringer committed:
python3Packages.scikit-learn: 0.24.1 -> 1.0.1

jonringer committed:
python3Packages.apsw: 3.33.0-r1 -> 3.36.0-r1

jonringer committed:
python3Packages.dask-mpi: 2.21.0 -> 2021.11.0

jonringer committed:
python3Packages.dask-ml: 1.9.0 -> 2021.11.30

jonringer committed:
python3Packages.distributed: fix build

jonringer committed:
python3Packages.huggingface-hub: add missing pyyaml dependency

jonringer committed:
python3Packages.datasets: 1.11.0 -> 1.16.1

jonringer committed:
python3Packages.datashader: disable failing tests, upstream notified

fabaff committed:
python3Packages.werkzeug: 2.0.1 -> 2.0.2

fabaff committed:
odoo: update dependencies

fabaff committed:
python3Packages.google-cloud-bigquery: disable failing tests

fabaff committed:
python3Packages.doc8: fix build and enable tests

dotlambda commented:
Webssh should be moved to `all-packages.nix`.

fabaff committed:
python3Packages.sanic: disable redirect tests

fabaff committed:
python3Packages.green: limit to three test cases

jonringer commented:
waiting for ofborg to go green

dotlambda commented:
So no #147524?

jonringer commented:
1 comment

jonringer commented:
@ofborg eval

fabaff commented:
Looks like that `matplotlib` is missing [`packaging`](https://github.com/matplotlib/matplotlib/blob/main/setup.py#L320) which became a requirement in 3.5.0.

Do we want to fix that here or later?

jonringer commented:
> Looks like that `matplotlib` is missing [`packaging`](https://github.com/matplotlib/matplotlib/blob/v3.5.0/setup.py) which became a requirement in 3.5.0.
> 
> Do we want to fix that here or later?

Here, doing the final touches before merging to staging

fabaff committed:
python3Packages.policy-sentry: 0.11.18 -> 0.11.19

fabaff committed:
python3Packages.matplotlib: update requirements

fabaff committed:
python3Packages.ase: 3.22.0 -> 3.22.1

fabaff committed:
python3Packages.datasette: 0.59 -> 0.59.4

fabaff committed:
python3Packages.django-configurations: fix build

fabaff committed:
python3Packages.django-formtools: fix build

fabaff committed:
python3Packages.easy-thumbnails: fix build

fabaff committed:
python3Packages.graphql-core: add pytest-asyncio

fabaff committed:
python3Packages.httpx-socks: add missing dependencies

fabaff committed:
python3Packages.pandas: disable failing test

fabaff committed:
python3Packages.seaborn: disable failing tests

fabaff committed:
python3Packages.thinc: 8.0.10 -> 8.0.13

fabaff committed:
python3Packages.rpcq: disable failing test

fabaff committed:
python3Packages.typical: disable failing test

trofi commented:
`staging` and `staging-next` fail to build `bpython`:

```
$ nix build -f. python3Packages.bpython -L
...
python3.9-bpython> unpacking source archive /nix/store/9d0qp9mpxsvlxs9wkqvmbsyb02vpx7ff-bpython-0.22.1.tar.gz
python3.9-bpython> source root is bpython-0.22.1
python3.9-bpython> setting SOURCE_DATE_EPOCH to timestamp 1636367222 of file bpython-0.22.1/setup.cfg
python3.9-bpython> patching sources
python3.9-bpython> applying patch /nix/store/vz4q2sd2g43ham91p2f5sal27yikks7v-clipboard-make-which-substitutable.patch
python3.9-bpython> can't find file to patch at input line 16
python3.9-bpython> Perhaps you used the wrong -p or --strip option?
```

Bisect landed on  c064af6e4a3786baa4ce1d294734cd23a90eceb0 `python3Packages.bpython: 0.21 -> 0.22.1`.

trofi commented:
Does `python3Packages.grpcio-status` fetch for you? On `staging` and `staging-next` it fails for me as:

```
$ nix build -f. python3Packages.grpcio-status -L
grpcio-status> trying https://files.pythonhosted.org/packages/source/g/grpcio-status/grpcio-status-1.43.0.tar.gz
grpcio-status>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
grpcio-status>                                  Dload  Upload   Total   Spent    Left  Speed
grpcio-status>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
grpcio-status> 100  8686  100  8686    0     0  54978      0 --:--:-- --:--:-- --:--:-- 54978
error: hash mismatch in fixed-output derivation '/nix/store/3pajxzh43idqywpp65gqcs72i17mix0d-grpcio-status-1.43.0.tar.gz.drv':
         specified: sha256-JVM8TWXX1ROmOPDEuIsrZnAOO/Q+aZWlWjvRXsC3eQI=
            got:    sha256-IXWQBvNqf/v/GH1BkfQRjActiqn6aCOhGq14QqPGzNA=
error: 1 dependencies of derivation '/nix/store/nl5jvlc1wygk4bnpiqmdxwhyfn6icvkg-python3.9-grpcio-status-1.43.0.drv' failed to build
```

jonringer commented:
> Does `python3Packages.grpcio-status` fetch for you? On `staging` and `staging-next` it fails for me as:
> 
> ```
> $ nix build -f. python3Packages.grpcio-status -L
> grpcio-status> trying https://files.pythonhosted.org/packages/source/g/grpcio-status/grpcio-status-1.43.0.tar.gz
> grpcio-status>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> grpcio-status>                                  Dload  Upload   Total   Spent    Left  Speed
> grpcio-status>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> grpcio-status> 100  8686  100  8686    0     0  54978      0 --:--:-- --:--:-- --:--:-- 54978
> error: hash mismatch in fixed-output derivation '/nix/store/3pajxzh43idqywpp65gqcs72i17mix0d-grpcio-status-1.43.0.tar.gz.drv':
>          specified: sha256-JVM8TWXX1ROmOPDEuIsrZnAOO/Q+aZWlWjvRXsC3eQI=
>             got:    sha256-IXWQBvNqf/v/GH1BkfQRjActiqn6aCOhGq14QqPGzNA=
> error: 1 dependencies of derivation '/nix/store/nl5jvlc1wygk4bnpiqmdxwhyfn6icvkg-python3.9-grpcio-status-1.43.0.drv' failed to build
> ```

it inherits from grpc, so unless there's an update for that packge as well, just updating the grpc package will fail

jonringer commented:
#151666
